lib/org/chromium/webidl/URLExtractor.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'URLExtractor',
+
+  requires: ['org.chromium.webidl.IDLFileContents'],
+
+  properties: [
+    {
+      name: 'excludeRegexp',
+      // documentation: 'Excludes all URLs matching this expression.',
+    },
+    {
+      name: 'includeRegexp',
+      // documentation: 'Only include URLs matching this expression.',
+    }
+  ],
+
+  constants: {
+    URL_REGEX: /https?:\/\/(www\.)?[-_a-zA-Z0-9.]{2,256}\.[a-z]{2,6}(\/[^?#, \n]*)?(\?[^#, \n]*)?/g,
```

Comment:
I've modified the RegExp slightly from the original implementation. This one performs the following:
`http` followed by optional `s`, followed by `://`, `www.` is optional.
Labels can be up to 256 characters (each individually is 63, but overall is 256) and currently matching `[-_a-zA-Z0-9.]`. Followed by mandatory `.` and suffix of 2-6 alpha characters.

- Paths and queries are matches in a similar manner as before